.github/workflows/publish.yml

```diff
@@ -0,0 +1,188 @@
+name: Publish
+
+on:
+  release:
+    types: [created]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  preflight:
+    name: Preflight
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v6
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: latest
+
+      - name: Setup node
+        uses: actions/setup-node@v6
+        with:
+          node-version: 24
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup dependencies
+        uses: ./.github/actions/setup-deps
+        with:
+          gh-token: ${{ secrets.GH_TOKEN }}
+
+      - name: Check versions
+        shell: bash
+        run: node scripts/check-versions.mjs
+
+      - name: Build
+        shell: bash
+        run: pnpm build
+
+      - name: Test
+        shell: bash
+        run: pnpm test
+
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - runner: macos-latest
+            name: darwin-arm64
+            target: arm64
+          - runner: macos-latest
+            name: darwin-x64
+            target: x64
+          - runner: ubuntu-24.04-arm
+            name: linux-arm64
+            target: arm64
+          - runner: ubuntu-latest
```

Comment:
The latest? I believe the platform we compile on usually binds to a glibc version and dictates the minimum glibc version. And compiling on ubuntu v24 would mean these binaries require at least glibc 2.39, which would exclude any earlier ubuntu versions (which are very commonly used). I thought we would want to compile on the absolute oldest OS versions we can possibly use; ubuntu v20 would give 2.31. In lmdb-js I have been compiling with the `node:18` container, which uses debian v11, which also uses glibc 2.31.
I believe most of our current containers use Ubuntu v22. However, IIRC AWS's linux is often using a much older glibc version.
Anyway, we could certainly try this and see if it works. Maybe the bindings are different here.

Reply:
I'm basing this entirely off what we use to build RocksDB: https://github.com/HarperFast/rocksdb-prebuilds/blob/main/.github/workflows/build.yml#L51-L62.

Ubuntu 20.04 LTS is no longer supported as of May 31, 2025.